<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LEDStrip</class>
 <widget class="QWidget" name="LEDStrip">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>928</width>
    <height>402</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_6">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QLabel" name="chip_type_label">
       <property name="text">
        <string>Chip Type:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="chip_type_combobox">
       <item>
        <property name="text">
         <string>WS2801</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>WS2811</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>WS2812</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_chip_type_note">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line_5">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Number of LEDs:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="box_num_led">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>320</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Frame Duration [ms]:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="box_frame_duration">
       <property name="minimum">
        <number>10</number>
       </property>
       <property name="maximum">
        <number>32768</number>
       </property>
       <property name="value">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_clock_frequency">
       <property name="text">
        <string>Clock Frequency [Hz]:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="box_clock_frequency">
       <property name="minimum">
        <number>10000</number>
       </property>
       <property name="maximum">
        <number>2000000</number>
       </property>
       <property name="singleStep">
        <number>100000</number>
       </property>
       <property name="value">
        <number>1666666</number>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QComboBox" name="channel_mapping_combobox">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <item>
        <property name="text">
         <string>RGB</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>RBG</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>BRG</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>BGR</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>GRB</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>GBR</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>RGBW</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>RGWB</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>RBGW</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>RBWG</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>RWGB</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>RWBG</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>GRWB</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>GRBW</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>GBWR</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>GBRW</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>GWBR</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>GWRB</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>BRGW</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>BRWG</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>BGRW</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>BGWR</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>BWRG</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>BWGR</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>WRBG</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>WRGB</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>WGBR</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>WGRB</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>WBGR</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>WBRG</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Color R:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="box_r">
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="value">
        <number>255</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>G:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="box_g">
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>B:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="box_b">
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>W:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="box_w">
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="brightness_label">
       <property name="text">
        <string>Brightness:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="brightness_slider">
       <property name="maximum">
        <number>31</number>
       </property>
       <property name="value">
        <number>31</number>
       </property>
       <property name="sliderPosition">
        <number>31</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="button_color">
       <property name="text">
        <string>Show Color</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="button_black">
       <property name="text">
        <string>Clear Color</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Speed:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="box_speed">
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="button_dot">
       <property name="text">
        <string>Show Moving Color Dot</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="button_gradient">
       <property name="text">
        <string>Show Moving Color Gradient</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>Intensity:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="gradient_intensity">
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>10</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_gradient_intensity">
       <property name="text">
        <string>10%</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Supply Voltage:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_voltage">
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_4">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>chip_type_combobox</tabstop>
  <tabstop>box_num_led</tabstop>
  <tabstop>box_frame_duration</tabstop>
  <tabstop>box_clock_frequency</tabstop>
  <tabstop>channel_mapping_combobox</tabstop>
  <tabstop>box_r</tabstop>
  <tabstop>box_g</tabstop>
  <tabstop>box_b</tabstop>
  <tabstop>box_w</tabstop>
  <tabstop>brightness_slider</tabstop>
  <tabstop>button_color</tabstop>
  <tabstop>button_black</tabstop>
  <tabstop>box_speed</tabstop>
  <tabstop>button_dot</tabstop>
  <tabstop>button_gradient</tabstop>
  <tabstop>gradient_intensity</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
